src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -621,6 +621,48 @@ public Iterable<EventBooking> findAllByEventIdAndStatus(final String eventId, @N
     }
   }
 
+  /**
+   * Find all bookings for events.
+   *
+   * @param eventIds - the event of interest.
+   * @return an iterable with all the events matching the criteria.
+   * @throws SegueDatabaseException - if an error occurs.
+   */
+  @Override
+  public Iterable<EventBooking> findAllByEventIds(final List<String> eventIds) throws SegueDatabaseException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT event_bookings.* FROM event_bookings JOIN users ON users.id=user_id WHERE event_id IN (");
+
+    // insert placeholders into the query - sb.append is adding new content to an existing string without creating
+    // new string objects each time  to an existing string builder
+    for (int i = 0; i < eventIds.size(); i++) {
+      if (i > 0) {
+        sb.append(", ");
+      }
+      sb.append("?");
+    }
+    sb.append(") AND NOT users.deleted");
+
+    try (Connection conn = ds.getDatabaseConnection();
+         PreparedStatement pst = conn.prepareStatement(sb.toString())) {
+
+      // Set values for each placeholder based on the eventIds list
+      for (int i = 0; i < eventIds.size(); i++) {
+        pst.setString(i + 1, eventIds.get(i));
+      }
+
+      try (ResultSet results = pst.executeQuery()) {
+        List<EventBooking> returnResult = Lists.newArrayList();
```

Comment:
Please prefer native code to external helpers like this, unless they truly add something of value, or make your code easier to read. That would reduce unnecessary coupling. For instance, this could be simply `new ArrayList<>()`